ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/QueryMask.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ * http://www.confluent.io/confluent-community-license
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.engine.rewrite;
+
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.parser.DefaultKsqlParser;
+import io.confluent.ksql.parser.SqlBaseBaseVisitor;
+import io.confluent.ksql.parser.SqlBaseParser.CreateConnectorContext;
+import io.confluent.ksql.parser.SqlBaseParser.SingleStatementContext;
+import io.confluent.ksql.parser.SqlBaseParser.StatementsContext;
+import io.confluent.ksql.parser.SqlBaseParser.TablePropertiesContext;
+import io.confluent.ksql.parser.SqlBaseParser.TablePropertyContext;
+import io.confluent.ksql.rest.entity.KsqlRequest;
+import io.confluent.ksql.util.ParserUtil;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.antlr.v4.runtime.misc.Interval;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.apache.commons.lang3.StringUtils;
+
+public class QueryMask {
+
+  private final String maskedQuery;
+
+  private static final ImmutableSet<String> ALLOWED_KEYS = ImmutableSet.of("connector.class");
+  private static final ImmutableSet<String> DISALLOWED_KEYS = ImmutableSet.of("connection.url",
+      "mode", "topic.prefix", "table.whitelist", "key");
+  private static final String MASKED_VALUE = "'[string]'";
+
+  public QueryMask(final KsqlRequest request) {
+    maskedQuery = getMaskedStatement(request.getKsql());
+    System.out.println("here");
+  }
+
+  public String getMaskedQuery() {
+    return maskedQuery;
+  }
+
+  private String getMaskedStatement(final String query) {
+    try {
+      final ParseTree tree = DefaultKsqlParser.getParseTree(query);
+      return new Visitor().visit(tree);
+    } catch (final Exception e) {
+      return maskConnectProperties(query);
+    }
+  }
+
+  private String maskConnectProperties(final String query) {
```

Comment:
can you add some unit tests for this method since it has some hard-to-read regex patterns? If that needs to change, having positive and negative examples that it catches will be really handy

Reply:
Yeah. I'll let @vcrfxia to have a pass to see if this is on right track before I add tests.